assets/prototype/domain/eventEditor/containers/mutations/useUpdateTicketMutation.js

```diff
@@ -20,7 +19,29 @@ const useUpdateTicketMutation = () => {
 	toaster.loading(loading, toasterMessage);
 	toaster.error(error);
 
-	return { updateTicket, loading, error };
+	const updateHandler = ({ description, name, price }) => {
+		if (description) {
+			return updateTicket({
+				variables: { input: { clientMutationId: 'xyz', id, description } },
+			});
+		}
+
+		if (name) {
+			return updateTicket({
+				variables: { input: { clientMutationId: 'xyz', id, name } },
+			});
+		}
+
+		if (price) {
+			return updateTicket({
+				variables: { input: { clientMutationId: 'xyz', id, price } },
+			});
+		}
```

Comment:
what about something **like** :

```js
let field = '';
let fieldName = ''
// set description 
field = description ? description : field;
fieldName = description ? 'description' : fieldName ;
// set name
field = name? name: field;
fieldName = name? 'name' : fieldName ;
// set price
field = price? price: field;
fieldName = price? 'price' : fieldName ;
if (field) {
    const data = { [fieldName]: field };
    return updateTicket({
        variables: { input: { clientMutationId: 'xyz', id, ...data } },
    });
}
```

just a suggestion,
but this only requires an extra two lines of code to add a new field, which we will eventually be doing.
As well, this logic could likely be extracted into a helper function so that you just feed in all of the incoming field props, and any that have values will get set in the returned data object.
Again, just a suggestion (ie: not a requirement) that could make this logic easier to manage as we add more things

Reply:
My initial implementation started with 2 fields and I thought that was acceptable, but it's not working well with more than 2 fields.
I liked this suggestion:
```
As well, this logic could likely be extracted into a helper function so that you just feed in all of the incoming field props, and any that have values will get set in the returned data object.
```

Reply:
I've added conditionally keys to that `variables` object as described here: https://medium.com/@mikeh91/conditionally-adding-keys-to-javascript-objects-using-spread-operators-and-short-circuit-evaluation-acf157488ede
***
but this will needs to be simplified somehow with ramda or lodash to enhance readability